action-helm-tools/dependencies.sh

```diff
@@ -16,10 +16,12 @@ prep_git() {
   git checkout -f -- .
   DEFAULT_BRANCH=$(git ls-remote --symref "https://github.com/${GITHUB_REPOSITORY}.git" HEAD | grep refs/heads | awk '{split($2, a, "/"); print a[3] }')
   git checkout "$DEFAULT_BRANCH"
-  git reset --hard "origin/$DEFAULT_BRANCH"
 
   # If branch already exists use that one otherwise create it
   git checkout "$BRANCH_NAME" 2>/dev/null || git checkout -b "$BRANCH_NAME"
+
+  # Make sure we are on latest and greatest
+  git reset --hard "origin/$DEFAULT_BRANCH"
```

Comment:
First we check out $BRANCH_NAME (ci-tools/helm-dependency-updater), then we do a hard reset to origin/$DEFAULT_BRANCH. I assume the reset will fail if the head of $BRANCH_NAME is not an ancestor of or equal to the head of $DEFAULT_BRANCH. So maybe we should do

1. Delete $BRANCH_NAME, if it exists
2. Create $BRANCH_NAME, based on $DEFAULT_BRANCH